libs/dh/esett/feature-outgoing-messages/src/lib/details.component.ts

```diff
@@ -72,96 +81,96 @@ import { DhResolveModalComponent } from './resolve.componet';
     TranslocoPipe,
     TranslocoDirective,
     VaterStackComponent,
+
     WATT_TABS,
     WATT_CARD,
     WATT_DRAWER,
     WattDatePipe,
     WattCodeComponent,
     WattButtonComponent,
     WattDescriptionListComponent,
-    WattDescriptionListItemComponent,
     WattValidationMessageComponent,
+    WattDescriptionListItemComponent,
+
     DhEmDashFallbackPipe,
     DhOutgoingMessageStatusBadgeComponent,
   ],
 })
-export class DhOutgoingMessageDrawerComponent {
-  private readonly apollo = inject(Apollo);
-  private readonly toastService = inject(WattToastService);
-  private readonly httpClient = inject(HttpClient);
-  private readonly modalService = inject(WattModalService);
+export class DhOutgoingMessageDetailsComponent {
+  private httpClient = inject(HttpClient);
+  private toastService = inject(WattToastService);
+  private modalService = inject(WattModalService);
+  private navigation = inject(DhNavigationService);
+  private outgoingMessageByIdDocumentQuery = lazyQuery(GetOutgoingMessageByIdDocument);
 
-  private subscription?: Subscription;
+  outgoingMessage = computed(
+    () => this.outgoingMessageByIdDocumentQuery.data()?.esettOutgoingMessageById
+  );
 
-  outgoingMessage: DhOutgoingMessageDetailed | undefined = undefined;
+  canResolve = computed(
+    () =>
+      this.outgoingMessage()?.documentStatus === DocumentStatus.Rejected ||
+      (this.outgoingMessage()?.documentStatus === DocumentStatus.AwaitingReply &&
+        dayjs(new Date()).diff(this.outgoingMessage()?.lastDispatched, 'hours') >= 12)
+  );
 
-  @ViewChild(WattDrawerComponent)
-  drawer: WattDrawerComponent | undefined;
+  drawer = viewChild.required(WattDrawerComponent);
 
-  closed = output<void>();
+  // Router param
+  id = input.required<string>();
 
   dispatchDocument = signal<string | undefined>(undefined);
   responseDocument = signal<string | undefined>(undefined);
 
-  get messageTypeValue(): string {
-    if (this.outgoingMessage?.calculationType) {
-      return translate(
-        'eSett.outgoingMessages.shared.calculationType.' + this.outgoingMessage?.calculationType
-      );
+  loading = this.outgoingMessageByIdDocumentQuery.loading;
+
+  get messageTypeValue() {
+    const calculationType = this.outgoingMessage()?.calculationType;
+    if (calculationType) {
+      return translate('eSett.outgoingMessages.shared.calculationType.' + calculationType);
     }
 
     return emDash;
   }
```

Comment:
suggestion: can be converted to `computed`

Reply:
Done